editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/PropertyInputDialog.java

```diff
@@ -0,0 +1,221 @@
+	private String propertyValue = null;
+
+	private List<AbstractCamelModelElement> propertyList = new ArrayList<>();
+	private Element inputElement;
+	private boolean isEditDialog = false;
+
+	/**
+	 * Dialog constructor.
+	 * 
+	 * @param parent the parent
+	 */
+	public PropertyInputDialog(Shell parent) {
+		super(parent);
+	}
+
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		setTitle("Property Details");
+		getShell().setText("Bean Property");
+		if (isEditDialog) {
+			setMessage("Edit name and value details for the property.");
+		} else {
+			setMessage("Specify name and value details for the new property.");
+		}
+		Composite area = new Composite(parent, SWT.NULL);
+		GridLayout gridLayout = new GridLayout(2, false);
+		area.setLayout(gridLayout);
+		area.setLayoutData(new GridData(GridData.FILL_BOTH));
+		propertyNameText = createLabelAndText(area, "Name*");
+		if (propertyName != null && !propertyName.trim().isEmpty()) {
+			propertyNameText.setText(propertyName);
+		}
+		propertyNameText.addModifyListener(input -> propertyName = propertyNameText.getText().trim());
+
+		propertyValueText = createLabelAndText(area, "Value*");
+		if (propertyValue != null && !propertyValue.trim().isEmpty()) {
+			propertyValueText.setText(propertyValue);
+		}
+		propertyValueText.addModifyListener(input -> propertyValue = propertyValueText.getText().trim());
+
+		return area;
+	}
+
+	@Override
+	protected Control createButtonBar(Composite parent) {
+		Control rtnControl = super.createButtonBar(parent);
+		getButton(IDialogConstants.OK_ID).setEnabled(validate());
+		setErrorMessage(null);
+		return rtnControl;
+	}
+
+	/**
+	 * @param parent parent composite
+	 * @param label string to put in label
+	 * @return reference to created Text control
+	 */
+	protected Text createLabelAndText(Composite parent, String label) {
+		new Label(parent, SWT.NONE).setText(label);
+		Text newText = new Text(parent, SWT.BORDER);
+		newText.setLayoutData(new GridData(SWT.FILL, SWT.NONE, true, false));
+		newText.addFocusListener(new FocusListener() {
+			@Override
+			public void focusGained(FocusEvent e) {
+				// ignore
+			}
+
+			@Override
+			public void focusLost(FocusEvent e) {
+				getButton(IDialogConstants.OK_ID).setEnabled(validate());
+			}
+		});
+		newText.addKeyListener(new KeyListener() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+				// ignore
+			}
+
+			@Override
+			public void keyReleased(KeyEvent e) {
+				getButton(IDialogConstants.OK_ID).setEnabled(validate());
+			}
+		});
+		return newText;
+	}
+
+	private boolean nameIsUnique(String propName, String newPropName) {
+		boolean namesMatch = propName.contentEquals(newPropName);
+		if (namesMatch) {
+			if (initialPropertyName != null) {
+				boolean isOldName = initialPropertyName.contentEquals(newPropName);
+				if (!isOldName) {
+					return false;
+				}
+			} else {
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	protected boolean validate() {
```

Comment:
As code is quite well splitted, I think that PropertyInputDialog.validate() can be tested unitly